Update intro and trademark details in the README (#18895)
opensearch-project/OpenSearch: README.md

## Welcome!

> OpenSearch is an open-source, enterprise-grade search and observability suite that brings order to unstructured data at scale.

Reviewer: nit - " a community driven, open-source"

> OpenSearch is a community-driven, open-source, enterprise-grade search and observability suite that brings order to unstructured data at scale.

Reviewer 2: ```suggestion
OpenSearch is an open-source, enterprise-grade search and observability suite that brings order to unstructured data at scale.
```

Reviewer 2: @getsaurabh02 I think that "community-driven" is implied by "open-source".
